lightning/src/routing/network_graph.rs

```diff
@@ -1170,6 +1190,74 @@ impl NetworkGraph {
 		self.update_channel_intern(msg, None, None::<(&secp256k1::Signature, &Secp256k1<secp256k1::VerifyOnly>)>)
 	}
 
+	/// Override a (directed) channel's individual properties without having a full update
+	/// This method should be used exclusively in conjunction with rapid graph sync, which is intended
+	/// for mobile devices.
+	///
+	/// `short_channel_id`: Channel to be updated
+	///
+	/// `flags`: Channel flags to set. Least significant bit serves as directional indicator
+	///
+	/// `mutations: Vec<ChannelMutationProperty>`: Array of properties to mutate
+	pub fn override_directional_channel_properties(&self, short_channel_id: u64, flags: u8, timestamp: u32, chain_hash: BlockHash, mutations: ChannelMutationProperties) -> Result<(), LightningError> {
```

Comment:
Given the use of `CustomChannelUpdate` in the docs, would it make sense to rename `ChannelMutationProperties` to that and include the remaining parameters as fields? Or maybe use the name `IncrementalChannelUpdate` given this is one variation of `CustomChannelUpdate`.

Reply:
hm, interesting point. In the docs, `CustomChannelUpdate` refers to the over-the-wire format that is a deviation from the BOLT7 spec, but hadn't really considered there to be a necessity for the internal representation of the delta to abide by that naming.

I think this point is somewhat related to your next one regarding where incremental and full updates are constructed and applied.

Reply:
obsolete now